src/Operators/derivatives.jl

```diff
@@ -0,0 +1,39 @@
+####
+#### Derivative operators
+####
+
+@inline ∂x_caa(i, j, k, grid, u) = δx_caa(i, j, k, grid, u) / Δx(i, j, k, grid)
+@inline ∂x_faa(i, j, k, grid, c) = δx_faa(i, j, k, grid, c) / Δx(i, j, k, grid)
+
+@inline ∂y_aca(i, j, k, grid, v) = δy_aca(i, j, k, grid, u) / Δy(i, j, k, grid)
+@inline ∂y_afa(i, j, k, grid, c) = δy_afa(i, j, k, grid, c) / Δy(i, j, k, grid)
+
+@inline ∂z_aac(i, j, k, grid, w) = δz_aac(i, j, k, grid, w) / ΔzF(i, j, k, grid)
+@inline ∂z_aaf(i, j, k, grid, c) = δz_aaf(i, j, k, grid, c) / ΔzC(i, j, k, grid)
+
+####
+#### Derivative operators acting on functions
+####
+
+@inline ∂x_caa(i, j, k, grid, f::F, args...) where F<:Function = (f(i+1, j, k, grid, args...) - f(i,   j, k, grid, args...)) / Δx(i, j, k, grid)
```

Comment:
These lines are cutoff on my github viewer --- may want to make them shorter.

Reply:
You should be able to scroll to the right on GitHub.

I agree these lines are much longer than any sane style guide would suggest using, but I think I have a good reason for breaking the line limit here: I wanted to align the definition of all the operators vertically so that we can easily see the similarities and differences between each definition. I think this makes the operators easier to understand as it highlights how similar they are, and also makes them easier to debug as any typo will be easy to spot.

Do you think this is a good enough reason?

Reply:
You're right, I can scroll.

I think your reason is good. Feel free to keep them.

In the past I have used alignment to build the operators, and then once I have good tests I break it into a few lines. However, it could make sense via your arguments to keep it as one line for future code readers.